test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,334 @@
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t>> reads;
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<size_t> seeds{0};
+    std::normal_distribution<> dis_error_count{static_cast<double> (simulated_errors_), standard_deviation};
+    std::uniform_real_distribution<double> probability(0.0, 1.0);
+
+    for (size_t i = 0; i < number_of_reads; ++i)
+    {
+        // TODO avoid mutating the same position multiple times
+        // simulate concrete error number or use normal distribution 
+        uint8_t simulated_errors = (standard_deviation == 0) ? simulated_errors_ : 
+                                                               abs(std::round(dis_error_count(gen)));
+        std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(ref) - read_length - simulated_errors);
+        size_t rpos = random_pos(gen);
+        std::vector<alphabet_t> read_tmp{std::ranges::begin(ref) + rpos,
+                                         std::ranges::begin(ref) + rpos + read_length + simulated_errors};
+
+        for (uint8_t j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(std::ranges::begin(read_tmp) + read_length, std::ranges::end(read_tmp));
+        reads.push_back(read_tmp);
+    }
+    return reads;
+}
+
+template <typename alphabet_t>
+auto generate_repeating_sequence(size_t const template_length = 5000,
+                                 size_t const repeats = 20,
+                                 double const template_fraction = 1,
+                                 size_t const seed = 0)
+{
+    std::vector<alphabet_t> seq_template = generate_sequence<alphabet_t>(template_length, 0, seed);
+
+    // copy substrings of length len from seq_template mutate and concatenate them
+    size_t len = std::round(template_length * template_fraction);
+    uint8_t simulated_errors = 5;
+    len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;
+    std::vector<std::vector<alphabet_t>> collection = generate_reads(seq_template, repeats, len, simulated_errors,
+                                                                     0.15, 0.15);
+
+    std::vector<alphabet_t> ref;
+    for(std::vector<alphabet_t> seq : collection)
+        ref.insert(std::ranges::end(ref), std::ranges::begin(seq), std::ranges::end(seq));
+
+    return ref;
```

Comment:
Can you try
```cpp
return (generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | std::view::join | std::ranges::to<std::vector<alphabet_t>);
```

and then remove creating `collection` and copying into `ref`?

Reply:
/home/sven/devel/seqan3/test/performance/search/search_benchmark.cpp: In function ‘std::vector<_RealType> generate_repeating_sequence(size_t, size_t, double, size_t)’:
/home/sven/devel/seqan3/test/performance/search/search_benchmark.cpp:134:119: error: parse error in template argument list
     return (generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | std::view::join | std::ranges::to<std::vector<alphabet_t>);

Reply:
```
return (generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | std::view::join | std::ranges::to<std::vector<alphabet_t>>);
```
It's a missing `>`

Reply:
/home/sven/devel/seqan3/test/performance/search/search_benchmark.cpp: In instantiation of ‘std::vector<_RealType> generate_repeating_sequence(size_t, size_t, double, size_t) [with alphabet_t = seqan3::dna4; size_t = long unsigned int]’:
/home/sven/devel/seqan3/test/performance/search/search_benchmark.cpp:171:96:   required from here
/home/sven/devel/seqan3/test/performance/search/search_benchmark.cpp:133:86: error: no match for ‘operator|’ (operand types are ‘std::vector<std::vector<seqan3::dna4> >’ and ‘const ranges::view::view<ranges::view::join_fn>’)
 ate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | std::view::join | std::ranges::to<std::vector<alphabet_t>>);

Reply:
This will work
```cpp
return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join;
```

The whole function then should look like
```cpp
template <typename alphabet_t>
std::vector<alphabet_t> generate_repeating_sequence(size_t const template_length = 5000,
                                                    size_t const repeats = 20,
                                                    double const template_fraction = 1,
                                                    size_t const seed = 0)
{
    std::vector<alphabet_t> seq_template = generate_sequence<alphabet_t>(template_length, 0, seed);

    // copy substrings of length len from seq_template mutate and concatenate them
    size_t len = std::round(template_length * template_fraction);
    uint8_t simulated_errors = 5;
    len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;

    return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join;
}
```